pycrostates/preprocessing/spatial_filter.py

```diff
@@ -0,0 +1,194 @@
+from typing import Union
+
+import numpy as np
+from mne import BaseEpochs, pick_info
+from mne.bem import _check_origin
+from mne.channels import find_ch_adjacency
+from mne.channels.interpolation import _make_interpolation_matrix
+from mne.io import BaseRaw
+from mne.io.pick import _picks_by_type
+from mne.parallel import parallel_func
+from mne.utils.check import _check_preload
+from numpy.typing import NDArray
+
+from .._typing import CHData
+from ..utils._checks import _check_n_jobs, _check_type, _check_value
+from ..utils._docs import fill_doc
+from ..utils._logs import logger, verbose
+
+
+@fill_doc
+@verbose
+def apply_spatial_filter(
+    inst: Union[BaseRaw, BaseEpochs, CHData],
+    ch_type: str = "eeg",
+    exclude_bads: bool = True,
+    origin: Union[str, NDArray[float]] = "auto",
+    n_jobs: int = 1,
+    verbose=None,
+):
```

Comment:
After a second look, a couple more comments.
- `_check_origin()` is now called twice, at the beginning and after info sub-selecting, thus with 2 different info.
- the argument `ch_type` is a rename of the common `picks` argument. I think it would be better to support a regular `'picks'` argument, this way we can:
  - add future support for non-EEG channels
  - apply the spatial filter to a subset of the EEG channels
  
But I do understand that this changes somehow the structure of the function since it requires to check that the selection of channels makes sense. WDYT?

Reply:
As if for now, we could remove the `ch_type` argument as the function only supports EEG channel type. 
In the case we later implement support for other channel types, the filter would still be applied to each channel type independently ( similar to `interpolate_bads`). In addition, It wouldn't really make sense to apply the filter on a subset of channels, as you will end up with an instance containing  interpolated and non interpolated channels. I think this is a really edgy case that could lead to errors if miss-used while users who would like to use this feature could use `inst.pick()` just before applying the filter. 

What do you think ?